### ISSUE TITLE
"more actions > delete" proceeds without confirmation

### ISSUE THREAD

**Issue by nameanyone** [Jermolene/TiddlyWiki5#2527]
Unlike "Delete this tiddler" in the EditToolbar, the "delete" in "more actions" dropdown does not ask for a confirmation.

To reproduce on http://tiddlywiki.com/, create a new tiddler first. The confirmation is there for pre-existing tiddlers.

**Comment by pmario** [comment]
As soon, as there is content in the text field, it does ask.

**Comment by Jermolene** [comment]
Thanks @nameanyone, the issue is indeed that no confirmation is given for deleting tiddlers with a missing or empty text field.

[This commit](e69e2c1c91002c296e1789532ca74286fae8d5a4) is the culprit, which was triggered by this discussion:

https://groups.google.com/d/topic/tiddlywikidev/V32urFRTiKQ/discussion

**Comment by nameanyone** [comment]
@Jermolene, my use case is existing tiddlers that have tags and other fields, but not text. handleDeleteTiddlerEvent should check for any user-created field, not jus text.

Your "Ths commit" link is invalid, BTW.

**Comment by Jermolene** [comment]
> @Jermolene, my use case is existing tiddlers that have tags and other fields, but not text. handleDeleteTiddlerEvent should check for any user-created field, not just text.

I agree, I'd be happy to merge a pull request. I'd be inclined to refactor the logic into a new `Tiddler.prototype.isModified()` method in tiddler.js.

> Your "Ths commit" link is invalid, BTW.

Thanks, fixed.

**Comment by nameanyone** [comment]
> I'd be happy to merge a pull request

I'll think about it. No promises  :smile: .
@pmario, feel free...

**Comment by nameanyone** [comment]
The button in `EditTemplate/controls` asks for confirmation even if there is nothing but title (i.e. the tiddler was previously saved with nothing but a title). That seems to me a proper behaviour. One might consider the title alone to be valuable information that they do not want to lose accidentally.

The button in  `Buttons/more-tiddler-actions` should behave the same. That's what I meant when I opened this issue.

Further, I think <kbd>Delete</kbd> button should not appear in a new tiddler at all. There is nothing to delete yet, only to cancel (discard) or save (confirm). <kbd>Cancel</kbd> is already doing what I expect — as soon as the title is changed from the initial "New Tiddler", it asks "Do you wish to discard changes to the tiddler…".

**Comment by pmario** [comment]
Still valid and may be labelled "improvement"

**Comment by pmario** [comment]
This issue was brought up at: https://talk.tiddlywiki.org/t/how-do-i-require-a-delete-confirmation-for-a-tiddler-with-no-text-field/12647

Since we do promote the usage of fields instead of data-tiddlers IMO we should give this issue a new chance.

We should also check if there are any fields. So we should check if the tiddler is "really" empty.

Related commit mentioned earlier by Jeremy is: [This commit](https://github.com/Jermolene/TiddlyWiki5/commit/e69e2c1c91002c296e1789532ca74286fae8d5a4)

**Comment by username1290** [comment]
bump
